src/templates/partials/lighttpd.hbs

```diff
@@ -38,19 +16,58 @@ ssl.privkey = "/path/to/private_key"
 ssl.pemfile = "/path/to/signed_cert_followed_by_intermediates"
  {{#if (minver "1.0.2" form.opensslVersion)}}
   {{#if (minver "1.1.0" form.opensslVersion)}}
-ssl.openssl.ssl-conf-cmd = ("MinProtocol" => {{#if (includes "TLSv1" output.protocols)}}"TLSv1"{{else if (includes "TLSv1.1" output.protocols)}}"TLSv1.1"{{else if (includes "TLSv1.2" output.protocols)}}"TLSv1.2"{{else}}"TLSv1.3"{{/if}})
+   {{#if (includes "TLSv1" output.protocols)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1")
+   {{else if (includes "TLSv1.1" output.protocols)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.1")
+   {{else if (includes "TLSv1.2" output.protocols)}}
+    {{#if (minver "1.4.78" form.serverVersion)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")
+    {{else}}
+     {{#unless (minver "1.4.56" form.serverVersion)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")
+     {{else}}
+#ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.2")  # lighttpd {{form.serverVersion}} TLS default
+     {{/unless}}
+    {{/if}}
+   {{else if (includes "TLSv1.3" output.protocols)}}
+    {{#unless (minver "1.4.78" form.serverVersion)}}
+ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.3")
+    {{else}}
+#ssl.openssl.ssl-conf-cmd = ("MinProtocol" => "TLSv1.3")  # lighttpd {{form.serverVersion}} TLS default
+    {{/unless}}
+   {{/if}}
   {{else}}
 ssl.openssl.ssl-conf-cmd = ("Protocol" => "ALL, -SSLv2, -SSLv3{{#unless (includes "TLSv1" output.protocols)}}, -TLSv1{{/unless}}{{#unless (includes "TLSv1.1" output.protocols)}}, -TLSv1.1{{/unless}}{{#unless (includes "TLSv1.2" output.protocols)}}, -TLSv1.2{{/unless}}")
   {{/if}}
+  {{#if (minver "1.4.68" form.serverVersion)}}
+   {{#if output.serverPreferredOrder}}
+ssl.openssl.ssl-conf-cmd += ("Options" => "+ServerPreference")
+   {{/if}}
+  {{else}}
 ssl.openssl.ssl-conf-cmd += ("Options" => "{{#if output.serverPreferredOrder}}+{{else}}-{{/if}}ServerPreference")
+  {{/if}}
   {{#if output.ciphers.length}}
+   {{#if (minver "1.4.68" form.serverVersion)}}
+    {{#if (includes "old" form.config)}}
+ssl.openssl.ssl-conf-cmd += ("CipherString" => "{{{join output.ciphers ":"}}}")
+    {{else}}
+
+# lighttpd TLS defaults are widely supported by clients and should be preferred
+# See https://wiki.lighttpd.net/Docs_SSL
+# Uncomment to better match the less restricted Mozilla {{form.config}} spec.
```

Comment:
Wording +cipher naming compatibility +deeplink to PFS docs:

```suggestion
# lighttpd uses secure TLS defaults, see https://wiki.lighttpd.net/Docs_SSL#Perfect-Forward-Secrecy-PFS
# If you need to match Mozilla {{form.config}} config exactly, uncomment below (OpenSSL compatibility only)
```

Reply:
> lighttpd uses secure TLS defaults

The lighttpd TLS defaults encompass more than simply the cipher list, so linking to PFS section is too specific for the intention of the statement.

> If you need to match Mozilla {{form.config}} config exactly, uncomment below (OpenSSL compatibility only)

"OpenSSL compatibility only" is true of ssl-config-generator.  lighttpd supports more than a handful of TLS libraries.  While the suggestion is truthful, it is terse and likely to be confusing to consumers.  I think it an overspecification of unnecessary information.  If you use it with a different lighttpd TLS module instead of lighttpd mod_openssl, then it will either work or it will not work.  In the latter case, lighttpd will emit an error, and the end-user can figure things out from there.  Alternatively, we could overspecify and point to the server-side-tls mapping of cipher names between various libraries, but I think that is better in the wiki and not exploded into generated configs.

Reply:
if it's an argument why the ciphers are commented out, a link to the default ciphers list is a convenience. it's unnecessary to just let people figure that out on their own where that info is.

notes for openssl applicability was iirc taken from similar place elsewhere, where rules were only provided for that implementation.

Reply:
>  (if you had a better heading to link, I'd happily do so, but it's all tucked away under PFS)

The lighttpd TLS docs will be updated for the Jan 2025 release which changes TLS defaults to `MinProtocol => "TLSv1.3"`.  An improper deep link might be broken, or might not longer point to the relevant portion of the document.

> The note for OpenSSL–only compatibility was adapted from the same line of the existing config ... only simplified for brevity (comment length) to fit with the rest of the instructions

That was not clear to me, and I am intimately familiar with the material, so your "simplification for brevity" was not helpful.